pkg/attestation/crafter/materials/oci_image.go

```diff
@@ -60,13 +62,72 @@ func (i *OCIImageCrafter) Craft(_ context.Context, imageRef string) (*api.Attest
 	repoName := remoteRef.Repository.String()
 	i.logger.Debug().Str("name", repoName).Str("digest", remoteRef.DigestStr()).Msg("container image resolved")
 
+	// Check if the signature tag exists for the image digest
+	signDigest := i.CheckForSignature(ref, descriptor)
+
+	containerImage := &api.Attestation_Material_ContainerImage{
+		Id: i.input.Name, Name: repoName, Digest: remoteRef.DigestStr(), IsSubject: i.input.Output,
+		Tag: ref.Identifier(),
+	}
+
+	// If the signature digest exists, add it to the material
+	if signDigest != "" {
+		containerImage.SignatureDigest = signDigest
+	}
+
 	return &api.Attestation_Material{
 		MaterialType: i.input.Type,
-		M: &api.Attestation_Material_ContainerImage_{
-			ContainerImage: &api.Attestation_Material_ContainerImage{
-				Id: i.input.Name, Name: repoName, Digest: remoteRef.DigestStr(), IsSubject: i.input.Output,
-				Tag: ref.Identifier(),
-			},
-		},
+		M:            &api.Attestation_Material_ContainerImage_{ContainerImage: containerImage},
 	}, nil
 }
+
+// CheckForSignature checks if the signature tag exists for the image digest.
+// It first checks for a tag with the signature digest and, if not found, uses the referrers API to check for related material.
+func (i *OCIImageCrafter) CheckForSignature(originalRef name.Reference, originalDesc *remote.Descriptor) string {
+	trimmedDigest := strings.TrimPrefix(originalDesc.Digest.String(), "sha256:")
+	newTag := fmt.Sprintf("sha256-%s.sig", trimmedDigest)
+
+	// Create new reference for the signature tag
+	newRef, err := name.NewTag(fmt.Sprintf("%s:%s", originalRef.Context().Name(), newTag))
+	if err != nil {
+		i.logger.Error().Err(err).Msg("failed to create new tag reference")
+		return ""
+	}
+
+	// Try fetching the signature tag
+	if desc, err := remote.Head(newRef); err == nil {
+		i.logger.Debug().Str("name", newRef.String()).Msg("image signature found")
+		return strings.TrimPrefix(desc.Digest.String(), "sha256:")
+	}
+
+	i.logger.Debug().Err(err).Str("name", newRef.String()).Msg("image signature not found")
+
+	// Use the referrers API to look for related materials if the tag is not found
+	return i.checkReferrersForSignature(originalRef, originalDesc)
+}
+
+// checkReferrersForSignature uses the referrers API to find a matching signature artifact.
+func (i *OCIImageCrafter) checkReferrersForSignature(originalRef name.Reference, originalDesc *remote.Descriptor) string {
+	ref := originalRef.Context().Digest(originalDesc.Digest.String())
+	index, err := remote.Referrers(ref, remote.WithAuthFromKeychain(i.keychain))
+	if err != nil {
+		i.logger.Error().Err(err).Msg("failed to fetch referrers")
```

Comment:
I'm wondering if we should log this as `Error`. This will be shown in the CLI output, right? being an optional feature, I think it would be more an Info (if we really want to show it), or just a Debug.